### PR TITLE
Remove AXRoleDescription fields

### LINKED ISSUE #11: AXAPI descriptions reference superclass role names
It appears the names have been used to map the roles.

For example, it's been noted for doc-acknowledgments that we have:

AXRoleDescription: 'region'

The result is that the role gets announced as 'acknowledgments' in Chrome and Firefox but as 'region' in Safari.

Core-AAM 1.2 no longer includes the AXRoleDescription fields, but instead includes this note:

> User agent should return a user-presentable, localized string value for the AXRoleDescription.

We should look at implementing the same as part of fixing this problem.

### PR DESCRIPTION
**Pull request by mattgarrish**
This pull request matches the DPUB AAM with the core by removing the AXRoleDescription fields and replacing them with a note about generating them.

@cookiecrook @avneeshsingh @clapierre @gregoriopellegrino  I know you guys discussed the role announcement issue at TPAC. Could you confirm that this change does not impact those discussions and the possible resolution?

Fixes #11

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/dpub-aam/pull/14.html" title="Last updated on Oct 20, 2022, 2:14 PM UTC (e8461fd)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/dpub-aam/14/e70542d...e8461fd.html" title="Last updated on Oct 20, 2022, 2:14 PM UTC (e8461fd)">Diff</a>

### CODE REVIEW

**Comment by clapierre** [comment]
Matt I don't see the note about generating the descriptions.  Maybe I missed it?

**Comment by mattgarrish** [comment]
> Maybe I missed it?

No, looks like it got swapped out with note 2, which doesn't apply to this document, when I did a merge of the main branch to make sure everything was up to date. It's the same note as here: https://w3c.github.io/core-aam/#ftn.note1

I'll get that updated shortly. I just noticed that the core spec also now links to the note. I'll see if I can find the code to do that, too. (I originally started this change last spring and then put it on hold while the other discussions were going on.)

**Comment by clapierre** [comment]
Looks great now Matt, thanks.

File: index.html [hunk 1]
```diff
@@ -883,9 +827,7 @@ <h3>Role Mapping Table</h3>
 								</td>
 								<td><p>Expose <code>ROLE_LINK</code> and object attribute
 											<code>xml-roles:doc-noteref</code>. </p></td>
-								<td>AXRole: <code>AXLink</code><br /> AXSubrole: <code>&lt;nil&gt;</code><br />
-									AXRoleDescription: <code>'link'</code>
-								</td>
+								<td>AXRole: <code>AXLink</code><br /> AXSubrole: <code>&lt;nil&gt;</code></td>
```
Comment (cookiecrook) [inline index.html:830]: We had agreed at TPAC to keep the link subroles, among others. Still investigating a way to add additional clarifiers, but this PR should probably be moved back to Draft until the corresponding ARIA issue is settled.

**Review by cookiecrook — CHANGES_REQUESTED** [review]

File: index.html (hunk 1 above)
Comment (cookiecrook) [reply index.html:830]: One of the proposals discussed was to expose the applied aria role through the API... Similar to how the DOM id is exposed.

**Comment by cookiecrook** [comment]
Current thought is that all existing AX API attributes would be retained, but a few new ones would be added, such as the raw string value of the role attribute, and where needed, some additional bikesheded property that contained the other user-presentable info, "bibliography reference"

**Comment by mattgarrish** [comment]
> Current thought is that all existing AX API attributes would be retained, but a few new ones would be added

Okay, in that case I'll close off this PR and issue #11.

Do you guys have a sense of how long it will take to find a solution to the announcement issue? We've been sitting on the 1.1 update for a few years now and it has some critical fixes like deprecating the list roles. If it's not a complicated fix, we can keep waiting, but if it's going to take some time to implement a solution would it make sense to move this to a 1.2 release?

**Comment by cookiecrook** [comment]
Still in discussion with the API owners, but thinking it will likely be with [AXCustomContent](https://developer.apple.com/documentation/accessibility/axcustomcontent).

**Pull request closed without merging**